packages/model-serving/src/components/deploymentWizard/fields/ModelLocationSelectField.tsx

```diff
@@ -35,124 +42,149 @@ export const isValidModelLocation = (value: string): value is ModelLocationField
-                    'opendatahub.io/connection-type': 'true',
-                  },
-                },
-                data: { fields: [] },
-              },
-              fieldValues: {},
-              additionalFields: {},
-            });
-            setSelectedConnection(undefined);
-          } else {
-            const foundConnectionType = modelServingConnectionTypes.find(
-              (ct) => ct.metadata.name === key,
-            );
+  }, [modelLocationData, modelServingConnectionTypes, selectedConnection]);
 
-            if (foundConnectionType) {
-              setModelLocationData({
-                type: ModelLocationType.NEW,
-                connectionTypeObject: foundConnectionType,
-                fieldValues: {},
-                additionalFields: {},
-              });
-            }
-          }
-        }}
-        onBlur={validationProps?.onBlur}
-        placeholder="Select model location"
-        value={
-          modelLocation === ModelLocationType.NEW && modelLocationData?.connectionTypeObject
-            ? modelLocationData.connectionTypeObject.metadata.name
-            : modelLocation
-        }
-        toggleProps={{ style: { minWidth: '450px' } }}
-      />
-      <ZodErrorHelperText zodIssue={validationIssues} />
-      {modelLocation && (
-        <ModelLocationInputFields
-          modelLocation={modelLocation}
-          connections={connections}
-          connectionTypes={modelServingConnectionTypes}
-          selectedConnection={selectedConnectionState}
-          setSelectedConnection={setSelectedConnection}
-          selectedConnectionType={selectedConnectionType}
-          setModelLocationData={setModelLocationData}
-          resetModelLocationData={resetModelLocationData}
-          modelLocationData={modelLocationData}
-        />
-      )}
-    </FormGroup>
+  const baseOptions = React.useMemo(
+    () => [
+      ...(connections.length > 0
+        ? [{ key: ModelLocationType.EXISTING, label: 'Existing connection' }]
+        : []),
+      ...(pvcs.data.length > 0 ? [{ key: ModelLocationType.PVC, label: 'Cluster storage' }] : []),
+      ...modelServingConnectionTypes.map((ct) => ({
+        key: ct.metadata.name,
+        label: ct.metadata.annotations?.['openshift.io/display-name'] || ct.metadata.name,
+        value: ModelLocationType.NEW,
+      })),
+    ],
+    [connections.length, pvcs.data.length, modelServingConnectionTypes],
+  );
+
+  const selectOptions = React.useMemo(() => {
+    if (baseOptions.length <= 1) {
+      // Placeholder to avoid auto selecting as different options load in (doesn't actually show in the dropdown)
+      return [
+        {
+          key: '__placeholder__',
+          label: 'Select model location',
+          isPlaceholder: true,
+          isDisabled: true,
+          optionKey: '__placeholder__',
+        },
+        ...baseOptions,
+      ];
+    }
+    return baseOptions;
+  }, [baseOptions]);
+  return (
+    <Form maxWidth="450px">
```

Comment:
I don't think we should use the Form component inside a field. It should be used in a Step so you can manage all the fields